### PR TITLE
CODE RUB: Update Packages & implement new SameExceptionAs for add Post

### PR DESCRIPTION
**Pull request by Mehdi-Aghaei**
@hassanhabib 
Dear Hassan,
I will be happy if you take a look at the test in Modify failing with the new SameException, I remembered that you discuss that with Chris in the video but I don't know the solution.
I downgrade the Acceptance project to .NET 6 because tests were failing.

### CODE REVIEW

**Comment by cjdutoit** [comment]
@Mehdi-Aghaei a downgrade of the test project should not be required.

Here is the things we changed (which you have done on PostServiceTests.Validations.Add):
1) Upgrade the Xeptions package to the latest version
2) On the PostServiceTests.cs we changed the Expression to:
```
        private static Expression<Func<Xeption, bool>> SameExceptionAs(Xeption expectedException) =>
            actualException => actualException.SameExceptionAs(expectedException);
```
3) On all the validation and exception tests the `await Assert.ThrowsAsync` has moved up from the `then` into the `when` and we assigned the outcome of the `await Assert.ThrowsAsync` to a variable i.e. 
```
            PostValidationException actualPostValidationException =
               await Assert.ThrowsAsync<PostValidationException>(
                   addPostTask.AsTask);
```
(Also note that `() => addPostTask.AsTask()` should also be written as just `addPostTask.AsTask` as above)

4) In the `then` block we added 
```
            actualPostValidationException.Should().BeEquivalentTo(
                expectedPostValidationException);
```
You will also need to do steps 3 & 4 on the below to complete the change for the Post tests

- PostServiceTests.Validations.RetrieveById.cs
- PostServiceTests.Validations.Modify
- PostServiceTests.Validations.RemoveById
- PostServiceTests.Exceptions.Add
- PostServiceTests.Exceptions.RetrieveById.cs
- PostServiceTests.Exceptions.RetrieveAll.cs
- PostServiceTests.Exceptions.Modify
- PostServiceTests.Exceptions.RemoveById

Hope this helps

**Comment by Mehdi-Aghaei** [comment]
@cjdutoit Thank you for the information.
I will start modifying the code and tag you again soon to take a look.

**Comment by Mehdi-Aghaei** [comment]
@cjdutoit  all Tests that you mention were modified.
_ShouldThrowValidationExceptionOnModifyIfStorageCreatedDateNotSameAsCreatedDateAndLogItAsync()_
still failling and the error is **Expected the inner exception data to have 1 item, but found 2**

**Comment by cjdutoit** [comment]
> @cjdutoit  all Tests that you mention were modified.
> 
> _ShouldThrowValidationExceptionOnModifyIfStorageCreatedDateNotSameAsCreatedDateAndLogItAsync()_
> 
> still failling and the error is **Expected the inner exception data to have 1 item, but found 2**

@Mehdi-Aghaei did you compare the expected data with the actual data to see what the difference is? 

I suspect the 1st is, that the dates is not the same and the 2nd is date is not recent.  On my phone atm. I can review that later today when home if you are still stuck.

(I will make the detail a bit better on that assertion)

**Comment by hassanhabib** [comment]
@Mehdi-Aghaei @cjdutoit there's stull some build issues in our work.

**Comment by Mehdi-Aghaei** [comment]
@hassanhabib 
we reached a stable build after modifications.
please take a look if it's possible.

File: Taarafo.Core.lutconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<LUTConfig Version="1.0">
```
Comment (cjdutoit) [inline Taarafo.Core.lutconfig:1]: @hassanhabib, I think this file extension should be added to the .gitignore

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**